### PR TITLE
[lock_cloud_machines] - retry in case of connection errors

### PR DESCRIPTION
**Pull request by GuyAfik**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
retry mechanism in case we have momentarily connection errors to gitlab

### CODE REVIEW

File: Tests/scripts/infrastructure_tests/lock_cloud_machines_test.py [hunk 1]
```diff
@@ -1,7 +1,41 @@
...
+    side_effect_responses = []
+    for response in responses:
+        if not isinstance(response, dict):
+            side_effect_responses.append(response)
+        else:
+            r = Response()
+            r._content = json.dumps(response).encode()
+            side_effect_responses.append(r)
+
+    requests_mocker = mocker.patch.object(requests, 'get', side_effect=side_effect_responses)
```
Comment (daryakoval) [inline Tests/scripts/infrastructure_tests/lock_cloud_machines_test.py:35]: Why not use `requests_mock` like we do in the integrations?

Comment (GuyAfik) [reply Tests/scripts/infrastructure_tests/lock_cloud_machines_test.py:35]: i couldn't find a way to use `side_effect` with `requests_mock`, i do not think it supports it.

if you know anything about it, dm me :)

**Review by daryakoval — APPROVED** [review]

**Comment by daryakoval** [comment]
Please re-trigger nightly once the xsiam build is fixed and add a link here.

**Comment by GuyAfik** [comment]
> Please re-trigger nightly once the xsiam build is fixed and add a link here.

successful nightly: https://code.pan.run/xsoar/content/-/jobs/29472946 (the xsiam build failed because of a modeling rule - un-related to script)

**Pull request merged**